emcc.py

```diff
@@ -2709,15 +2740,33 @@ def module_export_name_substitution():
   logger.debug('Private module export name substitution with ' + shared.Settings.EXPORT_NAME)
   src = open(final).read()
   final = final + '.module_export_name_substitution.js'
-  replacement = "typeof %(EXPORT_NAME)s !== 'undefined' ? %(EXPORT_NAME)s : {}" % {"EXPORT_NAME": shared.Settings.EXPORT_NAME}
+  if shared.Settings.MINIMAL_RUNTIME:
+    replacement = shared.Settings.EXPORT_NAME
+  else:
+    replacement = "typeof %(EXPORT_NAME)s !== 'undefined' ? %(EXPORT_NAME)s : {}" % {"EXPORT_NAME": shared.Settings.EXPORT_NAME}
   with open(final, 'w') as f:
     f.write(src.replace(shared.JS.module_export_name_substitution_pattern, replacement))
   save_intermediate('module_export_name_substitution')
 
 
+def generate_minimal_runtime_html(target, options, js_target, target_basename,
```

Comment:
The default path requires the presence of `{{{ SCRIPT }}}` in the shell file, which minimal runtime shell file does not have. It also appends PROXY_TO_WORKER, and injects things like Emterpreter, Wasm Memory initializer, inline separate-asm, precise_f32=2 script replace mods, inject arrayutils.js, base64utils.js, uriutils.js, do single_file, etc. none of which are items that are supported in MINIMAL_RUNTIME, so it seems cleanest to use a different function, rather than go through all of that and wrangle build flags and options so that none of the cases would be triggered.